Uchu.Core/Data/ThreadedList.cs

```diff
@@ -0,0 +1,207 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace Uchu.Core.Data
+{
+    public class ThreadedList<T> : IList<T>, IDisposable where T : class
```

Comment:
There is a warning on this line: `Rename Uchu.Core.Data.ThreadedList<T> to end in 'Collection'.`

Reply:
I guess I can make this `ThreadedCollection<T>`. Curious why this is a warning in C#.